packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -41,6 +56,62 @@ export default class ReconnectionManager {
     this.configure(meeting);
   }
 
+  /**
+   * Sets the iceState to connected and clears any disconnect timeouts and
+   * related timeout data within the iceState.
+   *
+   * @returns {undefined}
+   * @public
+   * @memberof ReconnectionManager
+   */
+  iceReconnected() {
+    if (this.iceState.disconnected) {
+      LoggerProxy.logger.log('ReconnectionManager:index#iceDisconnected --> ice has reconnected');
+
+      this.iceState.waitForReconnection.resolve();
+
+      if (this.iceState.timer.clearTimeout) {
+        this.iceState.timer.clearTimeout();
+      }
+
+      this.iceState.waitForReconnection = Promise.resolve();
+      this.iceState.disconnected = false;
+    }
+  }
+
+  /**
+   * Set the iceState to disconnected and generates a timeout that waits for the
+   * iceState to reconnect and then resolves. Rejects if the timeout duration
+   * is reached.
+   *
+   * @returns {Promise<undefined>}
+   * @public
+   * @memberof ReconnectionManager
+   */
+  iceDisconnected() {
+    if (!this.iceState.disconnected) {
+      LoggerProxy.logger.log('ReconnectionManager:index#iceDisconnected --> waiting for ice reconnect');
+
+      this.iceState.disconnected = true;
+
+      let resolution;
+
+      this.iceState.waitForReconnection = new Promise((resolve, reject) => {
+        this.iceState.timer = setTimeout(() => {
+          reject(new Error(`ice reconnection did not occur in ${this.iceState.timeoutDuration}ms`));
+        }, this.iceState.timeoutDuration);
+
+        resolution = resolve;
+      });
+
+      this.iceState.waitForReconnection.resolve = resolution;
+
+      return this.iceState.waitForReconnection;
```

Comment:
```suggestion
      return new Promise((resolve, reject) => {
        this.iceState.timer = setTimeout(() => {
          reject(new Error(`ice reconnection did not occur in ${this.iceState.timeoutDuration}ms`));
        }, this.iceState.timeoutDuration);

        this.iceState.waitForReconnection.resolve = resolve;
      });
```

Reply:
I was thinking you didn't need to keep a reference to the promise. I believe the promise will stay around because you are saving a reference to the `resolve` function of the promise and the timer has a reference to the `reject` function